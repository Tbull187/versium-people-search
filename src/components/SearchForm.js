import React from 'react';
import { Row, Col, InputGroup, FormControl, Button } from 'react-bootstrap';
import searchResults from './SearchResults';
import SearchResults from './SearchResults';

const SearchForm = ({ search, searchResults, loading }) => {
  
  return (
    <div>
    <Row>
      <Col>
        <InputGroup>
          <InputGroup.Prepend>
            <InputGroup.Text id="fist-name-label">First Name</InputGroup.Text>
          </InputGroup.Prepend>
          <FormControl
            placeholder="First Name"
            aria-label="First Name"
            aria-describedby="fist-name-label" 
          />
        </InputGroup>
      </Col>
      <Col>
        <InputGroup>
          <InputGroup.Prepend>
            <InputGroup.Text id="last-name-label">Last Name</InputGroup.Text>
          </InputGroup.Prepend>
          <FormControl
            placeholder="Last Name"
            aria-label="Last Name"
            aria-describedby="last-name-label" 
          />
        </InputGroup>
      </Col>
      <Col>
      <InputGroup>
          <InputGroup.Prepend>
            <InputGroup.Text id="state-label">State</InputGroup.Text>
          </InputGroup.Prepend>
          <FormControl
            placeholder="State"
            aria-label="State"
            aria-describedby="state-label" 
          />
        </InputGroup>      
      </Col>
      <Col>
        <Button 
          variant="primary"
          onClick={() => search('elon', 'musk', 'la')}>Search</Button>
      </Col>
    </Row>
    <div className="searchResults">
      <SearchResults searchResults={searchResults} loading={loading}/>
    </div>
    </div>
  )
}

export default SearchForm;