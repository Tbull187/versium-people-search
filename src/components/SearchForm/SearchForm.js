import React, { useState } from 'react';
import { Row, Col, InputGroup, FormControl, Button } from 'react-bootstrap';
import SearchResults from '../SearchResults/SearchResults';

const SearchForm = ({ searchResults, validateSearch, formErrors, loading }) => {
  const [ firstName, setFirstName] = useState('');
  const [ lastName, setLastName] = useState('');
  const [ geoState, setGeoState] = useState('');

  return (
    <>
    <Row className="btn-row" xs={1} lg={4}>
      <Col>
        <InputGroup className={formErrors.firstName ? "input-error" : ""}>
          <InputGroup.Prepend>
            <InputGroup.Text id="fist-name-label">First Name</InputGroup.Text>
          </InputGroup.Prepend>
          <FormControl
            placeholder="First Name"
            aria-label="First Name"
            aria-describedby="fist-name-label"
            value={firstName}
            onChange={e => setFirstName(e.target.value)}
            disabled={loading}
          />          
        </InputGroup>
        {formErrors.firstName && <div className="error-msg">First name must exist</div>}
      </Col>
      <Col>
        <InputGroup className={formErrors.lastName ? "input-error" : ""}>
          <InputGroup.Prepend>
            <InputGroup.Text id="last-name-label">Last Name</InputGroup.Text>
          </InputGroup.Prepend>
          <FormControl
            placeholder="Last Name"
            aria-label="Last Name"
            aria-describedby="last-name-label"
            value={lastName}
            onChange={e => setLastName(e.target.value)}
            disabled={loading}
          />          
        </InputGroup>
        {formErrors.lastName && <div className="error-msg">Last name must exist</div>}
      </Col>
      <Col>
        <InputGroup className={formErrors.state ? "input-error" : ""}>
          <InputGroup.Prepend>
            <InputGroup.Text id="state-label">State</InputGroup.Text>
          </InputGroup.Prepend>
          <FormControl
            placeholder="State"
            aria-label="State"
            aria-describedby="state-label"
            value={geoState}
            onChange={e => setGeoState(e.target.value)}
            disabled={loading}
          />
        </InputGroup>
        {formErrors.state && <div className="error-msg">Geographic state must exist</div>}
      </Col>
      <Col md="auto">
        <Button 
          variant="primary"
          onClick={() => validateSearch(firstName, lastName, geoState)}
          disabled={loading}
        >
          Search
        </Button>
      </Col>
    </Row>
    <SearchResults searchResults={searchResults} loading={loading}/>
    </>
  )
}

export default SearchForm;