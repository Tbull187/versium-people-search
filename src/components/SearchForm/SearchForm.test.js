import React from 'react';
import SearchForm from './SearchForm';
import renderer from 'react-test-renderer';
import merge from 'lodash/merge';

describe('SearchForm Component', () => {
  it('renders correctly with default props', () => {
    const component = renderer.create(<SearchForm {...defaultProps} />).toJSON();
    expect(component).toMatchSnapshot();
  })
  
  it('renders validation errors', () => {
    const props = merge(defaultProps, {formErrors: {
      firstName: true,
      lastName: true,
      state: true,
    }});
  
    const component = renderer.create(<SearchForm {...props} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  })
})

const defaultProps = {
  searchResults: null,
  validateSearch: () => console.log('validateSearch()'),
  formErrors: {
    firstName: false,
    lastName: false,
    state: false,
  },
  loading: false
};