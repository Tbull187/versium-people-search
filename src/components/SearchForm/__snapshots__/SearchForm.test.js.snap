// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm Component renders correctly with default props 1`] = `
Array [
  <div
    className="btn-row row row-cols-lg-4 row-cols-1"
  >
    <div
      className="col"
    >
      <div
        className="input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="fist-name-label"
          >
            First Name
          </span>
        </div>
        <input
          aria-describedby="fist-name-label"
          aria-label="First Name"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="First Name"
          value=""
        />
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="last-name-label"
          >
            Last Name
          </span>
        </div>
        <input
          aria-describedby="last-name-label"
          aria-label="Last Name"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="Last Name"
          value=""
        />
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="state-label"
          >
            State
          </span>
        </div>
        <input
          aria-describedby="state-label"
          aria-label="State"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="State"
          value=""
        />
      </div>
    </div>
    <div
      className="col-md-auto"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>,
  <div
    className="search-results"
  />,
]
`;

exports[`SearchForm Component renders validation errors 1`] = `
Array [
  <div
    className="btn-row row row-cols-lg-4 row-cols-1"
  >
    <div
      className="col"
    >
      <div
        className="input-error input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="fist-name-label"
          >
            First Name
          </span>
        </div>
        <input
          aria-describedby="fist-name-label"
          aria-label="First Name"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="First Name"
          value=""
        />
      </div>
      <div
        className="error-msg"
      >
        First name must exist
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="input-error input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="last-name-label"
          >
            Last Name
          </span>
        </div>
        <input
          aria-describedby="last-name-label"
          aria-label="Last Name"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="Last Name"
          value=""
        />
      </div>
      <div
        className="error-msg"
      >
        Last name must exist
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="input-error input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="state-label"
          >
            State
          </span>
        </div>
        <input
          aria-describedby="state-label"
          aria-label="State"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="State"
          value=""
        />
      </div>
      <div
        className="error-msg"
      >
        Geographic state must exist
      </div>
    </div>
    <div
      className="col-md-auto"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>,
  <div
    className="search-results"
  />,
]
`;

exports[`renders correctly 1`] = `
Array [
  <div
    className="btn-row row row-cols-lg-4 row-cols-1"
  >
    <div
      className="col"
    >
      <div
        className="input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="fist-name-label"
          >
            First Name
          </span>
        </div>
        <input
          aria-describedby="fist-name-label"
          aria-label="First Name"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="First Name"
          value=""
        />
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="last-name-label"
          >
            Last Name
          </span>
        </div>
        <input
          aria-describedby="last-name-label"
          aria-label="Last Name"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="Last Name"
          value=""
        />
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="state-label"
          >
            State
          </span>
        </div>
        <input
          aria-describedby="state-label"
          aria-label="State"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="State"
          value=""
        />
      </div>
    </div>
    <div
      className="col-md-auto"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>,
  <div
    className="search-results"
  />,
]
`;

exports[`renders correctly with default props 1`] = `
Array [
  <div
    className="btn-row row row-cols-lg-4 row-cols-1"
  >
    <div
      className="col"
    >
      <div
        className="input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="fist-name-label"
          >
            First Name
          </span>
        </div>
        <input
          aria-describedby="fist-name-label"
          aria-label="First Name"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="First Name"
          value=""
        />
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="last-name-label"
          >
            Last Name
          </span>
        </div>
        <input
          aria-describedby="last-name-label"
          aria-label="Last Name"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="Last Name"
          value=""
        />
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="state-label"
          >
            State
          </span>
        </div>
        <input
          aria-describedby="state-label"
          aria-label="State"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="State"
          value=""
        />
      </div>
    </div>
    <div
      className="col-md-auto"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>,
  <div
    className="search-results"
  />,
]
`;

exports[`renders validation errors 1`] = `
Array [
  <div
    className="btn-row row row-cols-lg-4 row-cols-1"
  >
    <div
      className="col"
    >
      <div
        className="input-error input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="fist-name-label"
          >
            First Name
          </span>
        </div>
        <input
          aria-describedby="fist-name-label"
          aria-label="First Name"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="First Name"
          value=""
        />
      </div>
      <div
        className="error-msg"
      >
        First name must exist
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="input-error input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="last-name-label"
          >
            Last Name
          </span>
        </div>
        <input
          aria-describedby="last-name-label"
          aria-label="Last Name"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="Last Name"
          value=""
        />
      </div>
      <div
        className="error-msg"
      >
        Last name must exist
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="input-error input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="state-label"
          >
            State
          </span>
        </div>
        <input
          aria-describedby="state-label"
          aria-label="State"
          className="form-control"
          disabled={false}
          onChange={[Function]}
          placeholder="State"
          value=""
        />
      </div>
      <div
        className="error-msg"
      >
        Geographic state must exist
      </div>
    </div>
    <div
      className="col-md-auto"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>,
  <div
    className="search-results"
  />,
]
`;
